### PR TITLE
Fix: The latest GA should be updated in the stable tag [ED-10526]

### PR DESCRIPTION
**Pull request by maork-elementor**

### CODE REVIEW

File: .github/workflows/update-readme-txt/action.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+runs:
+  using: "composite"
+  steps:
+      - name: Get & Update Stable Tag In Readme.txt
+        shell: bash
+        run: |
+              # Get the latest ga tag e.g. 3.13.2
+              if [[ "${{ inputs.CHANNEL }}" == "cloud" || "${{ inputs.CHANNEL }}" == "beta" || "${{ inputs.CHANNEL }}" == "dev" ]]; then
+                latest_tag=$(git ls-remote --tags | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1 | sed 's/^refs\/tags\/v//')
```
[the remark this answers is not in the document]
Comment (yotamselementor) [reply .github/workflows/update-readme-txt/action.yml:21]: Maybe add a comment explaining this as well

File: .github/workflows/update-readme-txt/action.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+              fi
+
+              # Get the latest ga tag e.g. 3.13.2 from the package version in case of channel ga
+              if [[ "${{ inputs.CHANNEL }}" == "ga" ]]; then
+                latest_tag=${{ inputs.PACKAGE_VERSION }}
+              fi
+
+              sed -i -E "s/Stable tag: .*/Stable tag: ${latest_tag}/g" $GITHUB_WORKSPACE/elementor/readme.txt
+
+              echo "readme.txt updated with Stable tag: ${latest_tag}"
```
Comment (yotamselementor) [inline .github/workflows/update-readme-txt/action.yml:31]: Print the readme.txt file for debug

File: .github/workflows/update-readme-txt/action.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+              
+              
+
```
Comment (yotamselementor) [inline .github/workflows/update-readme-txt/action.yml:36]:
```suggestion

```

File: .github/workflows/get-previous-release/action.yml
Comment (yotamselementor) [inline .github/workflows/get-previous-release/action.yml:1]: Why did you change this file?

**Review by yotamselementor — CHANGES_REQUESTED** [review]

File: .github/workflows/update-readme-txt/action.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+inputs:
+  CHANNEL:
+    required: true
+    description: 'The channel to get the version to (ga, cloud, beta).'
+  PACKAGE_VERSION:
+    required: true
+    description: 'The package version to get the version from.'
+
+
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/update-readme-txt/action.yml:12]: Done

File: .github/workflows/update-readme-txt/action.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      - name: Get & Update Stable Tag In Readme.txt
+        shell: bash
+        run: |
+              # Get the latest ga tag e.g. 3.13.2
+              if [[ "${{ inputs.CHANNEL }}" == "cloud" || "${{ inputs.CHANNEL }}" == "beta" || "${{ inputs.CHANNEL }}" == "dev" ]]; then
+                latest_tag=$(git ls-remote --tags | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1 | sed 's/^refs\/tags\/v//')
+              fi
+
+              # Get the latest ga tag e.g. 3.13.2 from the package version in case of channel ga
+              if [[ "${{ inputs.CHANNEL }}" == "ga" ]]; then
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/update-readme-txt/action.yml:25]: Done

File: .github/workflows/update-readme-txt/action.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+                latest_tag=${{ inputs.PACKAGE_VERSION }}
+              fi
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/update-readme-txt/action.yml:27]: Done

File: .github/workflows/update-readme-txt/action.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/update-readme-txt/action.yml:36]: Done

File: .github/workflows/update-readme-txt/action.yml (hunk 2 above)
Comment (maork-elementor) [reply .github/workflows/update-readme-txt/action.yml:31]: Done

File: .github/workflows/update-readme-txt/action.yml (hunk 3 above)
Comment (maork-elementor) [reply .github/workflows/update-readme-txt/action.yml:36]: Done

File: .github/workflows/get-previous-release/action.yml
Comment (maork-elementor) [reply .github/workflows/get-previous-release/action.yml:1]: To improve the logic & support with "v" and without.

File: .github/workflows/update-readme-txt/action.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/update-readme-txt/action.yml:21]: I added support for with "v" and without "v". (With the current PR code changes)

`git ls-remote --tags | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1 | sed 's/^refs\/tags\/v//'`

I tried the Commit suggestion, didn't get any result (directly on my Terminal).
You can run this line on your local and point it to Elementor remote for testing : 
(Add remote name after the `tags` option)

`git ls-remote --tags elementor | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1 | sed 's/^refs\/tags\/v//'`

File: .github/workflows/update-readme-txt/action.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (yotamselementor) [reply .github/workflows/update-readme-txt/action.yml:12]: No, its not

File: .github/workflows/update-readme-txt/action.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (yotamselementor) [reply .github/workflows/update-readme-txt/action.yml:30]: git ls-remote --tags | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1 | sed 's/^refs\/tags\/v//'

image

[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/update-readme-txt/action.yml]: You can give this a try : 

`git ls-remote --tags | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E '[0-9]+\.[0-9]+\.[0-9]+$' | sed 's/^refs\/tags\/v//' | sed 's/^refs\/tags\///' | sort -V | tail -n 1`

File: .github/workflows/get-previous-release/action.yml
Comment (maork-elementor) [reply .github/workflows/get-previous-release/action.yml:1]: Separated to Separated PR

File: .github/workflows/update-readme-txt/action.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/update-readme-txt/action.yml]: Sorry, now the change is really pushed.

File: .github/workflows/update-readme-txt/action.yml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+                # grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta": This series of commands excludes tags that contain the specified substrings "-rc", "-cloud", "-dev", and "-beta".
+                # awk '{print $2}': This command prints the second field (i.e., the tag name) from the output of the previous command.
+                # grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$': This command uses a regular expression to match tags that follow the semantic versioning pattern "vX.Y.Z", where X, Y, and Z are numbers.
+                # sort -V: This command sorts the matched tags in ascending order based on their version numbers.
+                # tail -n 1: This command selects the last (i.e., the highest) version number among the matched tags.
+                # sed 's/^refs\/tags\/v//': This command removes the leading "refs/tags/v" prefix from the selected tag name.
+
+                latest_tag=$(git ls-remote --tags | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E '[0-9]+\.[0-9]+\.[0-9]+$' | sed 's/^refs\/tags\/v//' | sed 's/^refs\/tags\///' | sort -V | tail -n 1)
```
Comment (yotamselementor) [inline .github/workflows/update-readme-txt/action.yml:29]: This fetches the same results for me, but I think its a bit simpler
```suggestion
                latest_tag=$(git ls-remote --tags | awk '{print $2}' | sed 's/^refs\/tags\/v//' | sed 's/^refs\/tags\///' | sort -V | grep -E '^[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1)
```

**Review by yotamselementor — CHANGES_REQUESTED** [review]

**Review by yotamselementor — APPROVED** [review]
